### PR TITLE
Add line-navigation to :ObsidianOpen (AdvancedURI)

### PR DESCRIPTION
**Pull request by gerryhernandez**
This PR adds a thin integration with the Obsidian Advanced URI plugin, allowing :ObsidianOpen to automatically scroll the Obsidian editor to the same line as NeoVim's.

See: [Obsidian Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri)

Note: scrolling in the Reading viewmode does not currently work (see [this issue](https://github.com/Vinzent03/obsidian-advanced-uri/issues/91)). The Obsidian editor will scroll correctly when it's in edit-mode.

### CODE REVIEW

**Comment by gerryhernandez** [comment]
I'm very unsure about the way I propagated the `.setup({...})` config/opts over to the `command` module. Could use a second opinion on that, for sure.

File: lua/obsidian/command.lua [hunk 1]
```diff
@@ -6,6 +6,17 @@ local util = require "obsidian.util"
...
+local opts = {
+  use_advanced_uri = nil,
+}
+
+--- Inject init/setup dependencies here (internal use)
+---
+---@param use_advanced_uri boolean|?
+command.setup = function(use_advanced_uri)
+  opts.use_advanced_uri = use_advanced_uri
+end
```
Comment (epwalsh) [inline lua/obsidian/command.lua:18]: I don't think you need all this because you can just access `client.opts.use_advanced_uri` directly.

Comment (gerryhernandez) [reply lua/obsidian/command.lua:18]: I had a feeling I was doing something not quite right. Give me a sec.

File: lua/obsidian/command.lua [hunk 2]
```diff
@@ -97,16 +97,24 @@ command.open = function(client, data)
 
   local encoded_vault = util.urlencode(vault_name)
   local encoded_path = util.urlencode(tostring(path))
+
+  local uri
+  if client.opts.use_advanced_uri then
```
Comment (gerryhernandez) [inline lua/obsidian/command.lua:102]: Here we go. Thanks for pointing it out.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `:ObsidianFollowLink` and companion function `util.cursor_on_markdown_link`
 - Added `:ObsidianLink` and `:ObsidianLinkNew` commands.
 - Added configuration option `disable_frontmatter` for frontmatter disabling
+- Added line-navigation to :ObsidianOpen via the Obsidian Advanced URI plugin
```
Comment (epwalsh) [inline CHANGELOG.md:15]:
```suggestion
- Added line-navigation to `:ObsidianOpen` via the Obsidian Advanced URI plugin
```

**Review by epwalsh — APPROVED** [review]
Awesome! Thanks for this

**Pull request merged**